src/main/java/de/tum/in/www1/artemis/domain/TextCluster.java

```diff
@@ -201,5 +210,9 @@ public String toString() {
         }
         return baos.toByteArray();
     }
+
+    public int openTextBlockCount() {
+        return (int) blocks.stream().filter(textBlock -> !textBlock.isAssessable()).count();
```

Comment:
I think this might have a quite big performance impact, as `TextBlock::isAssessable` might trigger Hibernate to fetch the `Result`. I think this should be explained in a comment and it is very important to fetch Results with the initial query to get all Clusters, OR make sure we can read it from a Cache.

Reply:
I  replaced the database call. Now everything is loaded in the inital query